bentoml/yatai/__init__.py

```diff
@@ -75,16 +75,39 @@ def ApplyDeployment(self, request, context=None):
                 request.deployment.namespace or self.default_namespace
             )
 
-            # TODO: validate deployment config
-
-            # create or update deployment spec record
-            self.deployment_store.insert_or_update(request.deployment)
+            deployment_orm = self.deployment_store.get(
+                request.deployment.name, request.deployment.namespace
+            )
+            if deployment_orm:
```

Comment:
this naming is confusing - this is not an orm object but a protobuf object, simply use:

```
previous_deployment = self.deployment_store.get(...)
```